File: Context/Entities/Campanign.cs

using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;


namespace Context.Entities
{
    public class Campanign
    {
        [Key]
        public int Id { get; set; }

        //[MaxLength(80)]
        public string Name { get; set; }

        //[MaxLength(200)]
        public string Description { get; set; }

        //[MaxLength(60)]
        public string Country { get; set; }

        //[MaxLength(60)]
        public string City { get; set; }

        //[MaxLength(100)]
        public string Address { get; set; }

        //[MaxLength(40)]
        public string Phone { get; set; }
        public string CreatedById { get; set; }

        [ForeignKey("CreatedById")]
        public ApplicationUser CreatedBy { get; set; }

        public string UpdatedById { get; set; }

        [ForeignKey("UpdatedById")]
        public ApplicationUser UpdatedBy { get; set; }

        public DateTime CreationDate { get; set; }

        public DateTime DateUpdate { get; set; }
    }
}

File: InventorySystem/wwwroot/js/campanign.js
let datatable

$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    datatable = $('#tblDatos').DataTable({
        "ajax": {
            "url": "/Administrator/Campanign/GetAll"
        },
        "columns": [
            { "data": "name" },
            { "data": "description" },
            { "data": "country" },
            { "data": "city" },
            { "data": "address" },
            { "data": "phone" },
            {
                "data": "createdBy",
                "render": function (data) {
                    if (data == null) {
                        return "Unassigned";
                    }
                    else {
                        return data;
                    }
                }, "width": "20%"
            },
            { "data": "creationDate" },
            {
                "data": "updatedBy",
                "render": function (data) {
                    if (data == null) {
                        return "Unassigned";
                    }
                    else {
                        return data;
                    }
                }, "width": "20%"
            },
            { "data": "dateUpdate" },
            {
                "data": "id",
                "render": function (data) {
                    return `
                        <div class="text-center">
                           <a href="/Administrator/Campanign/Upsert/${data}" class="btn btn-success text-white" style="cursor:pointer">
                              <i class="bi bi-pencil-square"></i>  
                           </a>
                           <a onclick=Delete("/Administrator/Campanign/Delete/${data}") class="btn btn-danger text-white" style="cursor:pointer">
                                <i class="bi bi-trash3-fill"></i>
                           </a> 
                        </div>
                    `;
                }, "width": "20%"
            }
        ]

    });
}

function Delete(url) {

    swal({
        title: "Are you sure about Delete the Campaign?",
        text: "This record cannot be recovered",
        icon: "warning",
        buttons: true,
        dangerMode: true
    }).then((borrar) => {
        if (borrar) {
            $.ajax({
                type: "DELETE",
                url: url,
                success: function (data) {
                    if (data.success) {
                        toastr.success(data.message);
                        datatable.ajax.reload();
                    }
                    else {
                        toastr.error(data.message);
                    }
                }
            });
        }
    });
}